### ISSUE TITLE
Add config option to treat warnings as errors

### ISSUE THREAD

**Issue by josteink** [mozilla/addons-linter#1014]
### Describe the problem and steps to reproduce it:

First create a web-extension `manifest.json` with invalid permissions (like `"tbas"`). Then run `web-ext lint`.
### What happened?

`web-ext lint` produced a warning, but still exited with return code `0`. 
### What did you expect to happen?

Have ability to add config-flag to treat warnings like errors.

This would basically mean rewriting [the following code](https://github.com/mozilla/addons-linter/blob/master/src/linter.js#L428-L430):

``` javascript
if (this.config.runAsBinary === true) {
  process.exit(this.output.errors.length > 0 ? 1 : 0);
}
```

Into:

``` javascript
if (this.config.runAsBinary === true) {
  var exitcode = this.output.errors.length > 0 ? 1 : 0;
  if (exitcode === 0 && this.config.warningAsError === true) {
    exitcode = this.output.warnings.length > 0 ? 1 : 0;
  }
  process.exit(exitcode);
}
```

(Probably associated documentation needs to be updated as well, but that's the core code changes required.)
